manifests/server/grant.pp

```diff
@@ -302,18 +327,63 @@
       $_togrant_object = join($_object_name, '"."')
       # Never put double quotes into has_*_privilege function
       $_granted_object = join($_object_name, '.')
+      $_schema = $_object_name[0]
+      $_relation = $_object_name[1]
     }
     default: {
       $_granted_object = $_object_name
       $_togrant_object = $_object_name
+      $_schema = $_object_name
+      $_relation = '%'
     }
   }
 
-  $_unless = $unless_function ? {
-      false    => undef,
-      'custom' => $custom_unless,
-      default  => "SELECT 1 WHERE ${unless_function}('${role}',
-                  '${_granted_object}', '${unless_privilege}')",
+  if ($dialect == 'redshift' and $_lowercase_role =~ /^group (.*)/) {
+    # Built-in functions such as has_table_privilege don't work on
+    # groups in Redshift at this writing. Similarly,
+    # information_schema role tables do not appear to be consistently
+    # kept up to date. As such, we have to dive into the low-level
+    # aclitem[] within pg_catalog.pg_class to find what we're looking
+    # for.
+    # 
+    # This only works with object types that cleanly map to
+    # information_schema, and is incompatible with complex permission
```

Comment:
can you give an example of a complex permission type that wouldn't work here, versus one that would?

Reply:
I'll reword this. Basically, ALL TABLES IN SCHEMA and ALL SEQUENCES IN SCHEMA will not work here (doubly so for the latter, as Redshift does not support sequences).